pkg/operatorsource/catalogsourceconfigbuilder.go

```diff
@@ -22,20 +28,37 @@ func (b *CatalogSourceConfigBuilder) CatalogSourceConfig() *v1alpha1.CatalogSour
 	return &b.object
 }
 
-// WithMeta sets TypeMeta and ObjectMeta accordingly.
-func (b *CatalogSourceConfigBuilder) WithMeta(namespace, name string) *CatalogSourceConfigBuilder {
+// WithTypeMeta sets TypeMeta of the CatalogSourceConfig object.
+func (b *CatalogSourceConfigBuilder) WithTypeMeta() *CatalogSourceConfigBuilder {
 	b.object.TypeMeta = metav1.TypeMeta{
 		APIVersion: fmt.Sprintf("%s/%s",
 			v1alpha1.SchemeGroupVersion.Group, v1alpha1.SchemeGroupVersion.Version),
 		Kind: v1alpha1.CatalogSourceConfigKind,
 	}
 
-	b.object.ObjectMeta = metav1.ObjectMeta{
-		Name:      name,
-		Namespace: namespace,
-		Labels:    map[string]string{DatastoreLabel: "true"},
+	return b
+}
+
+// WithNamespacedName sets name and namespace of the CatalogSourceConfig object.
+func (b *CatalogSourceConfigBuilder) WithNamespacedName(namespace, name string) *CatalogSourceConfigBuilder {
```

Comment:
I'm not opposed to this refactor, but is it necessary for this pr?

Reply:
Previously, I had `WithMeta` that added the label as well. I did not want to populate an object with a label to pass it to `client.Get`. So some refactor was necessary, I chose to break it down by typemeta, namespacedName and label.